src/gluonnlp/model/bert.py

```diff
@@ -17,105 +17,33 @@
-        super(BERTLayerNorm, self).__init__(epsilon=epsilon, in_channels=in_channels,
-                                            prefix=prefix, params=params)
-
-    def hybrid_forward(self, F, data, gamma, beta):
-        """forward computation."""
-        return F.LayerNorm(data, gamma=gamma, beta=beta, axis=self._axis, eps=self._epsilon)
-
-
-class BERTPositionwiseFFN(BasePositionwiseFFN):
-    """Structure of the Positionwise Feed-Forward Neural Network for
-    BERT.
-
-    Different from the original positionwise feed forward network
-    for transformer, `BERTPositionwiseFFN` uses `GELU` for activation
-    and `BERTLayerNorm` for layer normalization.
-
-    Parameters
-    ----------
-    units : int
-        Number of units for the output
-    hidden_size : int
-        Number of units in the hidden layer of position-wise feed-forward networks
-    dropout : float
-        Dropout probability for the output
-    use_residual : bool
-        Add residual connection between the input and the output
-    weight_initializer : str or Initializer
-        Initializer for the input weights matrix, used for the linear
-        transformation of the inputs.
-    bias_initializer : str or Initializer
-        Initializer for the bias vector.
-    prefix : str, default None
-        Prefix for name of `Block`s (and name of weight if params is `None`).
-    params : Parameter or None
-        Container for weight sharing between cells. Created if `None`.
-    activation : str, default 'gelu'
-        Activation methods in PositionwiseFFN
-    layer_norm_eps : float, default None
-        Epsilon for layer_norm
-
-    Inputs:
-        - **inputs** : input sequence of shape (batch_size, length, C_in).
-
-    Outputs:
-        - **outputs** : output encoding of shape (batch_size, length, C_out).
-    """
-
-    def __init__(self, units=512, hidden_size=2048, dropout=0.0, use_residual=True,
-                 weight_initializer=None, bias_initializer='zeros',
-                 prefix=None, params=None, activation='gelu', layer_norm_eps=None):
-        super(BERTPositionwiseFFN, self).__init__(units=units, hidden_size=hidden_size,
-                                                  dropout=dropout, use_residual=use_residual,
-                                                  weight_initializer=weight_initializer,
-                                                  bias_initializer=bias_initializer,
-                                                  prefix=prefix, params=params,
-                                                  # extra configurations for BERT
-                                                  activation=activation,
-                                                  use_bert_layer_norm=True,
-                                                  layer_norm_eps=layer_norm_eps)
-
-
-class BERTEncoder(BaseTransformerEncoder):
+class BERTEncoder(HybridBlock, Seq2SeqEncoder):
```

Comment:
Why should we inherit `Seq2SeqEncoder`

Reply:
It designates that the `Seq2SeqEncoder` API is implemented. It's not checked anywhere in the library though, so we could also remove it